lib/widgets/recent_dm_conversations.dart

```diff
@@ -138,3 +149,55 @@ class RecentDmConversationsItem extends StatelessWidget {
           ]))));
   }
 }
+
+class _NewDmButton extends StatefulWidget {
+  final VoidCallback onPressed;
+
+  const _NewDmButton({required this.onPressed});
+
+  @override
+  State<_NewDmButton> createState() => _NewDmButtonState();
+}
+
+class _NewDmButtonState extends State<_NewDmButton> {
+  bool _pressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final fabBgColor = _pressed ? designVariables.fabBgPressed : designVariables.fabBg;
+    final fabLabelColor = _pressed ? designVariables.fabLabelPressed : designVariables.fabLabel;
+
+    return InkWell(
+      onTap: widget.onPressed,
+      onTapDown: (_) => setState(() => _pressed = true),
+      onTapUp: (_) => setState(() => _pressed = false),
+      onTapCancel: () => setState(() => _pressed = false),
+      borderRadius: BorderRadius.circular(28),
+      child: Container(
+        padding: const EdgeInsetsDirectional.fromSTEB(16, 12, 20, 12),
+        decoration: BoxDecoration(
+          color: fabBgColor,
+          borderRadius: BorderRadius.circular(28),
+          boxShadow: [
+            // TODO: pressed state shadow variant
+            BoxShadow(
+              color: const Color(0xff2B0E8A).withValues(alpha: 0.4),
+              blurRadius: 16,
+              spreadRadius: 0,
+              offset: const Offset(0, 4)),
+          ]),
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Icon(Icons.add, size: 24, color: fabLabelColor),
+            const SizedBox(width: 8),
+            Text(
+              zulipLocalizations.newDmFabButtonLabel,
+              style: TextStyle(fontSize: 20, height: 24 / 20)
+                .merge(weightVariableTextStyle(context, wght: 500))
+                .copyWith(color: fabLabelColor)),
```

Comment:
It should be fine to just specify the color when constructing `TextStyle`, right?